src/codegen/ast_interpreter.cpp

```diff
@@ -153,6 +157,21 @@ void ASTInterpreter::addSymbol(InternedString name, Box* value, bool allow_dupli
     sym_table[name] = value;
 }
 
+void ASTInterpreter::setGenerator(Box* gen) {
+    assert(gen->cls == generator_cls);
+    this->generator = static_cast<BoxedGenerator*>(gen);
```

Comment:
Could you have these assert that these values weren't already set, just to make it clear that these are just for initialization?